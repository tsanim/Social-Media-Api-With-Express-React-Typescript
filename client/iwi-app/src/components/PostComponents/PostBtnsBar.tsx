import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHeart, faComment, faEdit, faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import PostBtnsBarProps from '../../interfaces/Components/PostComponents/PostBtnsBarProps.interface';

function PostBtnsBar(props: PostBtnsBarProps) {
    return (
        <div className={props.btnsDivClassName}>
            <div className="left">
                {
                    props.isLiked
                        ? <button onClick={props.handleDislikePost} className="liked"><FontAwesomeIcon icon={faHeart} /></button>
                        : <button onClick={props.handleLikePost}><FontAwesomeIcon icon={faHeart} /></button>
                }
                <button style={(props.areCommentsShown ? { color: "#8AAAE5" } : { color: 'inherit' })} onClick={props.handleShowComments}><FontAwesomeIcon icon={faComment} /></button>
            </div>

            <div className="right">
                {/* Check Post username whether is equal to logged in username so button for delete and edit can show */}
                {
                    (props.creatorId === localStorage.getItem('userId'))
                        ? (
                            <div>
                                <button onClick={props.handleShowModal}><FontAwesomeIcon icon={faTrashAlt} /></button>
                                <button onClick={props.handleShowEditForm}><FontAwesomeIcon icon={faEdit} /></button>
                            </div>
                        )
                        : (<div>
                            {
                                localStorage.getItem('role') === 'Admin'
                                    ? <button onClick={props.handleShowModal}><FontAwesomeIcon icon={faTrashAlt} /></button>
                                    : null
                            }
                        </div>)
                }
            </div>

        </div>
    )
}

PostBtnsBar.propTypes = {
    btnsDivClassName: PropTypes.string,
    isLiked: PropTypes.bool,
    handleDisLikePost: PropTypes.func,
    handleLikePost: PropTypes.func,
    areCommentsShown: PropTypes.bool,
    handleShowComments: PropTypes.func,
    creatorId: PropTypes.string,
    handleShowModal: PropTypes.func,
    handleShowEditForm: PropTypes.func,
}

export default PostBtnsBar;