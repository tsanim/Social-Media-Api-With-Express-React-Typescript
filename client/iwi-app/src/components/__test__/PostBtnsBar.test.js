import React from 'react';
import PostBtnsBar from '../PostComponents/PostBtnsBar';

import { shallow } from 'enzyme';

function postBtnsBarComponentSetup() {
    const props = {
        btnsDivClassName: '',
        isLiked: true,
        handleDisLikePost: jest.fn(),
        handleLikePost: jest.fn(),
        showComments: false,
        handleShowComments: jest.fn(),
        creatorId: '123',
        handleShowModal: jest.fn(),
        handleShowEditForm: jest.fn(),
    }

    window.localStorage.setItem('userId', '123');

    const enzymeWrapper = shallow(<PostBtnsBar {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('PostBtnsBar component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = postBtnsBarComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render 4 div tags', () => {
        const { enzymeWrapper } = postBtnsBarComponentSetup();

        expect(enzymeWrapper.find('div')).toHaveLength(4);
    });

    it('should render 4 buttons', () => {
        const { enzymeWrapper } = postBtnsBarComponentSetup();

        expect(enzymeWrapper.find('button')).toHaveLength(4);
    });
});
