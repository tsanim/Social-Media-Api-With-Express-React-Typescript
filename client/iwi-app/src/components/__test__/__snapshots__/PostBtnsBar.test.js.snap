// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBtnsBar component should render initial layout 1`] = `
Array [
  <div
    className=""
  >
    <div
      className="left"
    >
      <button
        className="liked"
        onClick={[MockFunction]}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f004",
                "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
              ],
              "iconName": "heart",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
      <button
        onClick={[MockFunction]}
        style={
          Object {
            "color": "inherit",
          }
        }
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f075",
                "M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z",
              ],
              "iconName": "comment",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </div>
    <div
      className="right"
    >
      <div>
        <button
          onClick={[MockFunction]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f2ed",
                  "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
                ],
                "iconName": "trash-alt",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
        <button
          onClick={[MockFunction]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  576,
                  512,
                  Array [],
                  "f044",
                  "M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z",
                ],
                "iconName": "edit",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>
    </div>
  </div>,
]
`;
